interfaces/IDeviceInfo.h

```diff
@@ -25,136 +25,315 @@
+            string deviceName /* @brief Host name */;
+            string cpuLoad /* @brief Current CPU load (percentage) */;
+            CPULoadAvgs cpuLoadAvg /* @brief CPU load average */;
+            string serialNumber /* @brief Device serial number */;
+            string time /* @brief Current system date and time */;
+        };
+
+        // @property
+        // @brief Retrieves SystemInfo
+        virtual Core::hresult SystemInfo(System& system /* @out */) const = 0;
+    };
+
+    /* @json 1.0.0 */
+    struct ISocketMetadata : virtual Core::IUnknown {
+        enum { ID = ID_SOCKET_METADATA };
+
+        virtual ~ISocketMetadata() override = default;
+
+        struct Socket {
+            uint32_t total /* @brief Number of total */;
+            uint32_t open /* @brief Number of open */;
+            uint32_t link /* @brief Number of link */;
+            uint32_t exception /* @brief Number of exception */;
+            uint32_t shutdown /* @brief Number of shutdown */;
+            uint32_t runs /* @brief Number of runs */;
+        };
+
+        // @property
+        // @brief Retrieves SocketInfo
+        virtual Core::hresult SocketInfo(Socket& socket /* @out */) const = 0;
+    };
+
+    /* @json 1.0.0 */
+    struct IAddressMetadata : virtual Core::IUnknown {
+        enum { ID = ID_ADDRESS_METADATA };
+
+        virtual ~IAddressMetadata() override = default;
+
+        struct Address {
+            string name /* @brief Interface name */;
+            string mac /* @brief Interface MAC address */;
+            string /* @opaque */ ip /* @brief An array of Interface IP address */;
         };
+        using IAddressIterator = RPC::IIteratorType<Address, ID_DEVICE_INFO_ADDRESSES>;
 
-        typedef RPC::IIteratorType<VideoOutput, ID_DEVICE_CAPABILITIES_VIDEO_OUTPUT> IVideoOutputIterator;
-        typedef RPC::IIteratorType<ScreenResolution, ID_DEVICE_CAPABILITIES_RESOLUTION> IScreenResolutionIterator;
-
-        virtual uint32_t VideoOutputs(IVideoOutputIterator*& videoOutputs /* @out */) const = 0;
-        virtual uint32_t DefaultResolution(const VideoOutput videoOutput /* @in */, ScreenResolution& defaultResolution /* @out */) const = 0;
-        virtual uint32_t Resolutions(const VideoOutput videoOutput /* @in */, IScreenResolutionIterator*& resolutions /* @out */) const = 0;
-        virtual uint32_t Hdcp(const VideoOutput videoOutput /* @in */, CopyProtection& hdcpVersion /* @out */) const = 0;
-        virtual uint32_t HostEDID(string& edid /* @out */) const = 0;
-        virtual uint32_t HDR(bool& supportsHDR /*@out*/) const = 0;
-        virtual uint32_t Atmos(bool& supportsAtmos /*@out*/) const = 0;
-        virtual uint32_t CEC(bool& supportsCEC /*@out*/) const = 0;
+        // @property
+        // @brief Retrieves AddressInfo
+        // @param addresses: An array of Interface address
+        virtual Core::hresult AddressInfo(IAddressIterator*& ip /* @out */) const = 0;
```

Comment:
Maybe for consistency, if possible, call these methods like the interface bu than without the I, so:
virtual Core::hresult AddressMetadata(Device& device /* @out */) const = 0;

Reply:
done